docs/git-commands.md

```diff
@@ -0,0 +1,101 @@
+- The repo cloned
+- Your fork remote added
+- The main repo remote added
+
+You can now begin to work and create your own PR against the main repo. To do this, you must create a branch. It is a good convention to give the branch a descriptive name, usually something descibing the work in the PR, keep this name short and too the point, example `add-contact-page`
+
+```
+git checkout -b <<desciptive-name-here>>
+```
+Now you are working on a new branch, when you are happy with your changes or just want to save the changes to the branch you can commit it, it is good practice to run the following command first to see what files have been changed by your work.
+```
+git status
+```
+If all the files listed are the ones you are happy with to commit to your branch run the following series of commands.
+```
+git add .
+git commit -m '<<describe the changes here>>'
+git push <<your-remote-name>> <<your-branch-name>>
+```
+Now you have pushed your changes on your branch to your remote, refer to [our tutorial](tutorial.md) on how to create a PR from here
+
+## Checking out other peoples PR's
+In order to check out someones PR you need to add that persons remote, this can be done with the following command, just replace the tags with the persons GitHub name
+```
+git remote add <<persons-name>> https://github.com/<<Persons-GitHub-userName>>/SchoolWebsite.git
+```
+Now that the person who created the PR's remote is added to your git project, you will need to fetch that person's branches, that can be done with the following command after you have added their remote
+```
+git fetch <<persons-name>>
+```
+Now you have access to the branches associated with that person's fork of the repo. If you run the following:
+```
+git branch -v
+```
+You will see a list of available branches to you locally, you will now need to check the PR on GitHub to see what branch the PR was created on, once you got the name you can enter the following
+```
+git checkout <<PR-branch-name>>
+```
+Now you can view and run the code of the pr locally. Once you are done you can checkout a different branch or if you wish to return to the branch you were on before you checked out the PR run the following command
+```
+git checkout -
+```
+## Rebasing your own PR
+So with multiple people working on a project the master branch (the original repo) can get ahead of the branch you are working on, so you will need to catch your work up to the master, this is known as Rebasing.
+
+This process is done by the following, first you need to checkout master, the master is the branch that is in line with the origin repo on GitHub, once you are on the master branch you will need to pull the changes down from GitHub. Now your master branch locally is uptodate with the github repo. Next, you need to check out the branch you want to rebase, once on this branch, you can rebase off your local master branch. So to do that you can run the following commands
+```
+git checkout master
+git pull origin
```

Comment:
Maybe it's better to specify pulling down from master here? `git pull origin master`